shared-libs/couch-request/test/couch-request.js

```diff
@@ -0,0 +1,186 @@
+      });
+      it(`Rejects (method: put): (notPlainObject == null)`, done => {
+        chai.expect(couch_request.put(notPlainObject)).to.eventually.be.rejectedWith(optionsErrorMsg)
+          .and.be.an.instanceOf(Error).notify(done);
+      });
+      it(`Rejects (method: delete): (notPlainObject == null)`, done => {
+        chai.expect(couch_request.delete(notPlainObject)).to.eventually.be.rejectedWith(optionsErrorMsg)
+          .and.be.an.instanceOf(Error).notify(done);
+      });
+      it(`Rejects (method: head): (notPlainObject == null)`, done => {
+        chai.expect(couch_request.head(notPlainObject)).to.eventually.be.rejectedWith(optionsErrorMsg)
+          .and.be.an.instanceOf(Error).notify(done);
+      });
+    } else {
+      const toString = `${notPlainObject.toString()}`;
+      const result = toString === '' ? Object.getPrototypeOf(notPlainObject).constructor.name : toString;
+      it(`Rejects notPlainObject (method: get): ${result}`, done => {
+        chai.expect(couch_request.get(notPlainObject)).to.eventually.be.rejectedWith(optionsErrorMsg)
+          .and.be.an.instanceOf(Error).notify(done);
+      });
+      it(`Rejects notPlainObject (method: post): ${result}`, done => {
+        chai.expect(couch_request.post(notPlainObject)).to.eventually.be.rejectedWith(optionsErrorMsg)
+          .and.be.an.instanceOf(Error).notify(done);
+      });
+      it(`Rejects notPlainObject (method: put): ${result}`, done => {
+        chai.expect(couch_request.put(notPlainObject)).to.eventually.be.rejectedWith(optionsErrorMsg)
+          .and.be.an.instanceOf(Error).notify(done);
+      });
+      it(`Rejects notPlainObject (method: delete): ${result}`, done => {
+        chai.expect(couch_request.delete(notPlainObject)).to.eventually.be.rejectedWith(optionsErrorMsg)
+          .and.be.an.instanceOf(Error).notify(done);
+      });
+      it(`Rejects notPlainObject (method: head): ${result}`, done => {
+        chai.expect(couch_request.head(notPlainObject)).to.eventually.be.rejectedWith(optionsErrorMsg)
+          .and.be.an.instanceOf(Error).notify(done);
+      });
+    }
+  });
+});
+
+describe('Couch request receives correct options and returns stub value', () => {
+
+  let requestGet;
+  let requestPost;
+  let requestPut;
+  let requestDelete;
+  let requestHead;
+
+  beforeEach(() => {
+    sinon.stub(process, 'env').value({ ...process.env, COUCH_URL: 'http://admin:password@test.com:5984/medic' });
+    requestGet = sinon.stub(request, 'get');
+    requestPost = sinon.stub(request, 'post');
+    requestPut = sinon.stub(request, 'put');
+    requestDelete = sinon.stub(request, 'delete');
+    requestHead = sinon.stub(request, 'head');
+    requestGet.returns(Promise.resolve('foo'));
+    requestPost.returns(Promise.resolve('foo'));
+    requestPut.returns(Promise.resolve('foo'));
+    requestDelete.returns(Promise.resolve('foo'));
+    requestHead.returns(Promise.resolve('foo'));
```

Comment:
It's better to provide the response you want in the test not the setup - right now if the code accidentally called `get` instead of `post` the test would pass because they're all resolving the same thing!

Reply:
I'll have a look.